components/raftstore-v2/src/worker/tablet.rs

```diff
@@ -371,28 +380,57 @@ impl<EK: KvEngine> Runner<EK> {
             self.background_pool
                 .spawn(async move {
                     // sync flush for leader to let the flush happend before later checkpoint.
-                    tablet.flush_cfs(DATA_CFS, true).unwrap();
-                    let elapsed = now.saturating_elapsed();
-                    // to be removed after when it's stable
-                    info!(
-                        logger,
-                        "flush memtable for leader";
-                        "region_id" => region_id,
-                        "duration" => ?elapsed,
-                    );
-
+                    if threshold.is_none() || tablet.has_old_active_memtable(threshold.unwrap()) {
+                        // tablet.flush_cfs(DATA_CFS, true).unwrap();
+                        let r = tablet.flush_cfs(DATA_CFS, true);
+                        let elapsed = now.saturating_elapsed();
+                        if let Err(e) = r {
+                            warn!(
+                                logger,
+                                "flush memtable for leader failed";
+                                "region_id" => region_id,
+                                "reason" => reason,
+                                "err" => ?e,
+                            );
+                            return;
+                        } else {
+                            info!(
+                                logger,
```

Comment:
Will this log be too annoying given the fact we have many flush triggered by manual flush or split/merge.

Reply:
Manual flush doesn't go this path. Logs can be removed if merge preflush works as expected.